### ISSUE TITLE
Support additional chai matchers for BigNumber

### ISSUE THREAD

**Issue by hbarcelos** [EthWorks/Waffle#347]
It would be nice to have additional matchers that already work for plain numbers:

- `closeTo`
- `within`

Then a test like this:

```js
    it("Should return the correct task price before submission timeout", async () => {
      const expectedPrice = taskMinPrice.add(taskMaxPrice.sub(taskMinPrice).mul(secondsPassed).div(submissionTimeout));
      const delta = expectedPrice.div(100);

      const actualPrice = await contract.getTaskPrice(taskId, task);

      expect(actualPrice.sub(expectedPrice).abs()).to.be.lt(delta, "Invalid task price");
    });
```

could be written like:

```js
    it("Should return the correct task price before submission timeout", async () => {
      const expectedPrice = taskMinPrice.add(taskMaxPrice.sub(taskMinPrice).mul(secondsPassed).div(submissionTimeout));
      const delta = expectedPrice.div(100);

      const actualPrice = await contract.getTaskPrice(taskId, task);

      expect(actualPrice).to.be.closeTo(expectedPrice, delta, "Invalid task price");
    });
```

**Comment by marekkirejczyk** [comment]
Hi @hbarcelos,
Thank you for the issue.
I think it is worth considering, would love to hear more opinions @sz-piotr.
I wonder why would you need delta, if contracts are deterministic and simple by design, so that you can just put the exact number. Would love to hear more.

**Comment by hbarcelos** [comment]
Hi there @marekkirejczyk.

Some of the products and services here at Kleros leverage a price discovery feature to help users to find the best price.

This can be better illustrated by this chart:
![Price Discovery](https://user-images.githubusercontent.com/16565602/93901519-f9a3a200-fccc-11ea-91c4-c39947c9869c.png)

Instead of providing a fixed price, users provide a min and a max price they are willing to pay for a good or service which has some sort of deadline.

Starting from the point the contract call is made, the actual price increases linearly until the deadline. When some service provider or seller believes the price is fair, they accept it and the remaining balance is transferred back to the user.

In order to properly test those contracts, we need delta-matching, otherwise we'd be tightly coupled with how the underlying test provider deals with time and blocks.

**Comment by marekkirejczyk** [comment]
Sounds good to me. Would you like to prepare a PR?

**Comment by hbarcelos** [comment]
I will be able to do so in about a week or 2, it that's okay.

**Comment by marekkirejczyk** [comment]
Sounds great!

**Comment by naddison36** [comment]
+1 for this feature.

My use case is I want to check a `timestamp` emitted in an event is `within` a few seconds of the current time

**Comment by marekkirejczyk** [comment]
Fixed by #416 and released in 3.3.0
